integration/spark/src/main/common/java/io/openlineage/spark/api/OpenLineageContext.java

```diff
@@ -53,6 +54,12 @@
    */
   @NonNull OpenLineage openLineage;
 
+  /**
+   * A non-null, but potentially empty, list of {@link
+   * io.openlineage.spark.agent.facets.EnvironmentFacet } to be used to get environment
+   */
+  @NonNull Properties properties;
```

Comment:
With https://github.com/OpenLineage/OpenLineage/pull/512 merged, I think we can eliminate this property and simply trigger the facet builder on `SparkListenerJobStart` events, right? Given that the `properties` here are a copy of the actual SparkContext's thread local properties, thus is subject to change from one job start event to the next, I don't think we ought to keep a copy here. It is likely to confuse facet builder authors who may think this field is kept up to date, when really it isn't (the context is only created once at the beginning of the sql job - subsequent RDD jobs will _not_ update the properties object in the context).